test/parallel_testsuite.py

```diff
@@ -119,14 +129,16 @@ def __init__(self, max_cores, options):
     self.max_failures = options.max_failures
     self.failing_and_slow_first = options.failing_and_slow_first
     self.progress_counter = 0
+    self.use_single_line_output = options.buffer and sys.stderr.isatty()
+    if self.use_single_line_output:
+      self.terminal_width = shutil.get_terminal_size()[0]
```

Comment:
How expensive is getting this size? Caching this size here sounds like things will break if one then resizes the window size mid-run?

It might be quite common for user to start a test run, then mid-run realize "the logs are not fitting on one line, let me grow the terminal window a bit."